File: Java/26. Remove Duplicates from Sorted Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> numberSet = new HashSet<>();
        int n = 0;
        for (int i = 0; i < nums.length; i++) {
            if (!numberSet.contains(nums[i])) {
                numberSet.add(nums[i]);
                nums[n++] = nums[i];
            }
        }
        return n;
    }
}

File: Cpp/26. Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        std::set<int> numberSet;
        int n = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (numberSet.find(nums[i]) == numberSet.end()) {
                numberSet.insert(nums[i]);
                nums[n++] = nums[i];
            }
        }
        return n;
    }
};
